test/es-module/test-importmap.mjs

```diff
@@ -0,0 +1,203 @@
+      /^Error \[ERR_INVALID_IMPORT_MAP\]: Invalid import map: top level key "imports" is required and must be a plain object$/
+    );
+  });
+
+  it('resolve - empty importmap', async () => {
+    const importMap = await getImportMap('empty');
+    const spec = importMap.resolve('test');
+    assert.strictEqual(spec, 'test');
+  });
+
+  it('resolve - simple importmap', async () => {
+    const importMap = await getImportMap('simple');
+    assert.strictEqual(
+      importMap.resolve('foo').pathname,
+      new URL('node_modules/foo/index.mjs', entryPoint).pathname
+    );
+    assert.strictEqual(
+      importMap.resolve('bar', new URL('node_modules/foo/index.mjs', entryPoint)).pathname,
+      new URL('baz.mjs', entryPoint).pathname
+    );
+    assert.strictEqual(importMap.resolve('bar'), 'bar');
+  });
+
+  it('resolve - nested scopes', async () => {
+    const importMap = await getImportMap('unordered-scopes');
+    assert.strictEqual(
+      importMap.resolve('zed', new URL('node_modules/bar', entryPoint)).pathname,
+      new URL('node_modules/bar/node_modules/zed/index.mjs', entryPoint).pathname
+    );
+    assert.strictEqual(
+      importMap.resolve('zed', new URL('node_modules/bar/node_modules/zed', entryPoint)).pathname,
+      new URL('baz.mjs', entryPoint).pathname
+    );
+  });
+
+  it('resolve - data url', async () => {
+    const importMap = await getImportMap('dataurl');
+    assert.strictEqual(
+      importMap.resolve('foo').href,
+      'data:text/javascript,export default () => \'data\''
+    );
+  });
+
+  it('should pass --experimental-import-map', async () => {
+    const importMapPath = fixtures.path('es-module-loaders/importmaps/simple.json');
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--experimental-import-map', importMapPath,
+      entryPoint.pathname,
+    ], {
+      cwd: fixtures.path('es-module-loaders/importmaps'),
+    });
+
+    assert.strictEqual(code, 0, stderr);
+    assert.strictEqual(stdout, 'baz\n');
+    assert.strictEqual(signal, null);
+  });
+
+  it('should throw on startup on invalid import map', async () => {
+    const importMapPath = fixtures.path('es-module-loaders/importmaps/invalid.json');
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--experimental-import-map', importMapPath,
+      entryPoint.pathname,
+    ], {
+      cwd: fixtures.path('es-module-loaders/importmaps'),
+    });
+
+    assert.strictEqual(code, 1);
+    assert.strictEqual(signal, null);
+    assert.strictEqual(stdout, '');
+    assert(stderr.includes('Invalid import map: top level key "imports" is required'));
+  });
+
+  it('should handle import maps with absolute paths', async () => {
+    const importMapPath = path.resolve(tmpdir.path, 'absolute.json');
+    await writeFile(importMapPath, JSONStringify({
+      imports: {
+        foo: fixtures.path('es-module-loaders/importmaps/node_modules/foo/index.mjs'),
```

Comment:
Shouldn't it be URLs in a import map?
```suggestion
        foo: fixtures.fileURL('es-module-loaders/importmaps/node_modules/foo/index.mjs'),
```

Reply:
I should also add, this is just writing the static json, so all that matters is that the output is *not* `file://` but the simple absolute path.

I will go through on another pass and see if I can simplify these tests, but I don't think we are well served by being pedantic on the tests, they are intended to IMO to cover real expected input from end users.

Reply:
Ok, simplified again but did not change the written files to include `file://` as I explicitly do not think we should require that (and I don't think the spec says so either).